### ISSUE TITLE
handleJoinRequest seems to work on an ad hoc basis

### ISSUE THREAD

**Issue by indianbhaji** [sentanos/roblox-js-server#3]
Hi,

I'm having problems with handleJoinRequest. It seems to work when it intends to, and otherwise not work when it's needed.

I'm using the entire script you have provided from the examples.

Here's the script which executes the function in the ModuleScript:

```
local age = plr.AccountAge
	
-- server info
local server = require(script.Parent.Server)
local domain = ''
local key = ''
	
-- user info
local groupId = 2660996
	
-- exec api
local api = server(domain, key, groupId)
	
if age > 45 then
	print((api.handleJoinRequest(plr.Name, 'true')).message)
else
	print((api.handleJoinRequest(plr.Name, 'false')).message)
end
```

I've got my domain filled out, and I've set the key to the one I've inserted in `settings.json`. Otherwise, I'm not very sure as to what's exactly going wrong here.

The error I keep on getting is `Handle join request failed: No join request was found with that username`.

I've got FilteringEnabled turned on, if that's making a difference. The input is from a ScreenGui, and when a player clicks on a button, it fires the event in Workspace, triggering the function in the ModuleScript which should in theory, accept the member.

**Comment by sentanos** [comment]
Could you clarify, when does it work and when does it not work?

**Comment by indianbhaji** [comment]
I can't answer that, and that's the problem.

**Comment by sentanos** [comment]
Does it never work or does it work sometimes?

**Comment by indianbhaji** [comment]
It works sometimes, I don't know why.

**Comment by sentanos** [comment]
I may have had this issue before, I will look into it.

**Comment by sentanos** [comment]
I have confirmed the problem but am extremely puzzled as to why it is occurring.

**Comment by sentanos** [comment]
I've messed around with ROBLOX's API for hours and can only conclude that the problem is due to the steaming piece of shit ROBLOX calls a website. Their pages are totally inconsistent, the search url especially works sometimes but doesn't work others: on some computers it will work less often than other, some headers seem to help while others don't, and the entire thing is totally random.

_However_, I have edited handleJoinRequest to make it a bit more reliable. It should now search the initial request, which works more often than the search does (although sometimes it still doesn't). For the most part this will work for you unless you have a lot of join requests, in which it will be forced to use the search API. It will also tell you if this was the problem, with the error message being "Couldn't load search API". If you receive this error you should just send handleJoinRequest again and keep doing so until it eventually works.

And again it really does matter what system you are using it on, I just seriously can't figure out why. On my main computer it worked nearly every time but on my pi it was the opposite. Unfortunately on heroku the search url pretty much never loads.

Because it is handled by my dependency roblox-js, I have fixed the problem there in [`roblox-js/357c2da`](https://github.com/sentanos/roblox-js/commit/357c2dace6b3b05e56c479f946346137c3fdd308) and you don't have to change any specific server files, just the dependency. If you are using heroku follow [these](https://github.com/sentanos/roblox-js-server/issues/2#issuecomment-264600153) instructions to update it.

I'm going to continue to look into this. Although again handleJoinRequest should work for you, it is still very unstable.

**Comment by indianbhaji** [comment]
@sentanos That's great! I'll keep this issue open, for the time being, I'll use that. If you have any better system, let me know!

**Comment by sentanos** [comment]
I've finally figured it out. Turns out ROBLOX only creates the temporary URL on the server you request the group admin page (by IP address) on. Because different computers have different DNS caching systems, sometimes it will be the same when you go to the next request and sometimes it will be different.

I've fixed this completely in [`roblox-js/5c4a849`](https://github.com/sentanos/roblox-js/commit/5c4a84955a85ff96ae3c3eb20da3ae1874e714c3) which means after updating your server dependencies (as I explained in my previous comment) handleJoinRequest will work correctly on the server.

Please confirm a final time that everything is working correctly and then close the issue, thanks for reporting.

**Comment by indianbhaji** [comment]
Seems work fine, thank you very much!
